doc/unwrap.md

```diff
@@ -1,20 +1,30 @@
 % What #[derive(Unwrap)] generates
 
-When an enum is decorated with `#[derive(Unwrap)]`, for each variant `foo` in the enum,
-with fields `(a, b, c, ...)` a public instance method `unwrap_foo(self) -> (a, b, c, ...)`
-is generated. If you don't want the `unwrap_foo` method generated for a variant,
-you can put the `#[unwrap(ignore)]` attribute on that variant.
+When an enum is decorated with `#[derive(Unwrap)]`, for each variant `foo` in the enum, with fields `(a, b, c, ...)` a public instance method `unwrap_foo(self) -> (a, b, c, ...)` is generated.
+If you don't want the `unwrap_foo` method generated for a variant, you can put the `#[unwrap(ignore)]` attribute on that variant.
+If you want to treat a reference, you can put the `#[unwrap(ref)]` attribute on the enum declaration or that variant, then `unwrap_foo_ref(self) -> (&a, &b, &c, ...)` will be generated. You can also use mutable references by putting `#[unwrap(ref_mut)]`.
 
 # Example usage
 
 ```rust
 # #[macro_use] extern crate derive_more;
 
 #[derive(Unwrap)]
+#[unwrap(ref)]
 enum Maybe<T> {
     Just(T),
     Nothing,
 }
+
+fn main() {
+    assert_eq!(Maybe::Just(1).unwrap_just(), 1);
+
+    // Panics if variants are different
+    // assert_eq!(Maybe::Nothing.unwrap_just(), /* panic */);
+    // assert_eq!(Maybe::Just(2).unwrap_nothing(), /* panic */);
+
+    assert_eq!(*(&Maybe::Just(42)).unwrap_just_ref(), 42);
```

Comment:
The instant dereference looks a bit weird in the example. This looks like more common code I think:
```suggestion
    assert_eq!(&Maybe::Just(42)).unwrap_just_ref(), &42);
```